tests/differencers.cc

```diff
@@ -0,0 +1,146 @@
+#include "common.h"
+
+// BoB robotics includes
+#include "imgproc/mask.h"
+#include "navigation/differencers.h"
+
+// Standard C++ includes
+#include <algorithm>
+#include <utility>
+
+using namespace BoBRobotics;
+using namespace BoBRobotics::Navigation;
+
+#define COMPARE_MASK(m1, m2, expected, expectedMean, mask1, mask2)                    \
+    do {                                                                              \
+        ASSERT_EQ(m1.type(), CV_8UC1);                                                \
+        ASSERT_EQ(m2.type(), CV_8UC1);                                                \
+        ASSERT_EQ(expected.type(), CV_8UC1);                                          \
+        ASSERT_EQ(m1.size(), m2.size());                                              \
+        bool doesMatch;                                                               \
+        float mean;                                                                   \
+        std::tie(doesMatch, mean) = compare<AbsDiff>(m1, m2, expected, mask1, mask2); \
+        EXPECT_TRUE(doesMatch);                                                       \
+        EXPECT_TRUE(mean == expectedMean);                                            \
+    } while (false);
```

Comment:
The idiomatic version of the ``do{}while(false)`` macro trick leaves off the trailing semicolon meaning that it _needs_ to be called with a semicolon afterwards (like a _real_ function)

Reply:
Ah, good spot!